Графика для всех страниц

TromiUss opened the pull request:
непонятные моменты прокомментировал в коде ( 1 ) и в коммитах

---
:mortar_board: [Разметка всех страниц](https://up.htmlacademy.ru/htmlcss/36/user/2300497/tasks/5)

---
:mortar_board: [Графика для всех страниц](https://up.htmlacademy.ru/htmlcss/36/user/2300497/tasks/7)

TromiUss committed:
header

Добавил часть header

TromiUss committed:
header-full

Полное описание header

TromiUss committed:
хиро-блок

Пока не до конца понял, как отобразить векторную графику, поэтому добавил текстом

TromiUss committed:
исправил ком-й

TromiUss committed:
...

TromiUss committed:
H1

С заголовком определился

TromiUss committed:
section-hero

Посмотрел запись лайва, исправил секцию "hero"

TromiUss committed:
секция преимущества

добавил преимущества, какие-то с картинкой, поэтому добавил тег <img>

TromiUss committed:
преимущества текст

Не совсем понял, как именно нужно оформлять эту часть, поэтому сделал просто заголовок с текстом

Как такового описания нет, поэтому <p> поставил без класса

TromiUss committed:
советы

добавил секцию с советами и картинками ( векторные )

TromiUss committed:
секция с поиском

секция с формой на подбор отеля
кнопку не описывал, тк хочу реализовать Popup

на stackoverflow, конечно, нашел решение, но до 8 части пока не дошли, поэтому не совсем понятен принцип этой функции

TromiUss committed:
форма подписки

TromiUss committed:
форма подписки

Добавил <img>

TromiUss committed:
последний шаг

добавил оставшуюся часть/=.
векторное изображение-ссылку сделал через <img>

TromiUss committed:
fix footer

пару исправлений
ссылка на сайт html academy и кликабельные соцсети

TromiUss committed:
Update index.html

TromiUss committed:
Update index.html

efiand commented:
Множество коммитов хорошо, когда по ним идеально распределены фрагменты кода или отдельные файлы. Когда одна строка расползается на несколько коммитов, такое довольно сложно ревьюировать - и если комп не дымится, есть смысл не коммитить до завершения работ.

efiand commented on index.html:
```diff
@@ -5,8 +5,26 @@
-    <p>Репозиторий создан для обучения на профессиональном онлайн‑курсе «<a href="https://htmlacademy.ru/intensive/htmlcss">HTML и CSS. Профессиональная вёрстка сайтов</a>».</p>
-
+    <header class="header-page">
+      <a href="#" class="header-page link">
```
Практически невозможно адекватно работать со стилями в будущем, когда одним и тем же классом назван **и** хедер, **и** ссылка в нём. Абсолютно все стили шапки, включая сетку, будут применены к ссылке, ведь у нее такой же класс. И придется писать чуть ли не вдвое больше кода в костыльный селектор `.page-header.link`, чтобы отменять на ссылке все стили самой шапки. Потом - то же самое для всех стилей ниже.
Проще задать разным сущностям разные классы - например, `header-page-link, header-page-navigation` и т. п.

Я не буду комментировать все подобные случаи, ваша задача - найти их и сделать так, чтобы у элементов разного уровня не было одинаковых классов.

efiand commented on index.html:
```diff
@@ -5,8 +5,26 @@
+      <nav class="header-page navigation">
+        <ul class="header-page list">
+          <li class="header-page item">
+            <a href="#">Главная</a>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-03

> Для ссылок, адрес которых неизвестен, можно использовать атрибут href с решёткой href="#"

Адрес этой ссылки известен - `index.html`. Можно задать его, можно убрать href, так как это и есть та же страница.

> Ссылки, которые ведут на самих себя, **могут** быть без атрибута.

Как лучше, до сих пор спорят, выбор за вами.

efiand commented on index.html:
```diff
@@ -5,8 +5,26 @@
+            <a href="#">Главная</a>
+          </li>
+          <li class="header-page item">
+            <a href="About-Sedona.html">О седоне</a>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#PROJ-03

> Файлы и папки **должны быть** записаны строчными буквами без пробелов между словами.

Здесь есть прописные буквы.

---

https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-03
> Для ссылок, адрес которых неизвестен, можно использовать атрибут href с решёткой href="#"

У нас нет такой страницы, поэтому следует указывать `#` или `#!`.

efiand replied:
Ссылки-заглушки рекомендую писать как `<a href="#!">` по двум причинам:

- Отсутствует скачок к верхней части страницы, что важно для всего, что после хедера, но на проекте все должно быть единообразно, поэтому и тут стоит так сделать.
-  При замене временных ссылок на реальные под замес не попадет реальная ссылка к верху страницы, если она есть на проекте.

efiand commented on index.html:
```diff
@@ -21,6 +21,25 @@
+            <a href="#" class="navigation-link">
+              <span class="visually-hidden">
+                Поиск
+            </span>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#PROJ-01

Форматирование должно быть единообразным.
Сколько отступов слева от открываюшегося тега, столько и от закрывающегося.

Не буду комментировать остальные подобные проблемы, нужно отформатировать всю страницу корректно.
Можно в VS Code использовать клавиши `alt + shift + f` для автоформатирования

efiand commented on index.html:
```diff
@@ -21,6 +21,25 @@
         </ul>
+        <ul class="header-page user-list">
+          <li class="header-page item">
+            <a href="#" class="navigation-link">
```
Желательно соблюдать порядок атрибутов, например, class в начале, это упрощает анализ кода при внесении правок и доработок.
Если вы используете emmet, его поведение в VS Code (чтобы class был спереди) можно скорректировать через настройки:

![изображение](https://user-images.githubusercontent.com/26799424/215143761-07c8921d-f42e-4250-93fa-1ff5954d96c3.png)
```json
  "emmet.preferences": {
    "output.reverseAttributes": true
  },
```

efiand commented on index.html:
```diff
@@ -21,6 +21,25 @@
+          <li class="header-page item">
+            <a href="#" class="navigation-link">
+              <span class="visually-hidden">
+                Поиск
```
https://up.htmlacademy.ru/htmlcss/36/criteries#ALLY-02

> Подпись в конце описания **должна** иметь точку. Точка нужна для интонационной паузы при чтении скринридером.

(аналогичные случаи в других местах найдите самостоятельно, не буду дублировать комментарии)

efiand commented on index.html:
```diff
@@ -42,16 +42,16 @@
-      <p class="main-hero bottom-text">Because the Grand Canyon sucks!</p>
-    -->
+    <main class="main-index">
+      <section class="hero">
```
Здесь не какая-то часть основного контента, а вводная часть, и тег `<header>`, на мой взгляд, подошел бы больше.

efiand commented on index.html:
```diff
@@ -42,16 +42,16 @@
-    -->
+    <main class="main-index">
+      <section class="hero">
+        <h2 class="visually-hidden">О Седона</h2>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML

> На странице должен быть `<h1>`, который описывает содержимое страницы.
Уровни заголовков должны идти последовательно от большего к меньшему без пропусков.

Можно этот заголовок переделать в `<h1>`

efiand commented on index.html:
```diff
@@ -42,16 +42,16 @@
+        <img src="#" alt="Пейзаж-гор">
+        <img src="#" alt="Welcome to the gorgeous Sedona, because the Grand Canyon sucks!">
+      </section>
+      <!-- Если с векторной не получится, то сделаю через <span> -->
```
Если с векторной не получится, обращайтесь, подскажу, как сделать, потому что должно получаться :-)

efiand commented on index.html:
```diff
@@ -42,16 +42,16 @@
+      <section class="hero">
+        <h2 class="visually-hidden">О Седона</h2>
+        <img src="#" alt="Пейзаж-гор">
+        <img src="#" alt="Welcome to the gorgeous Sedona, because the Grand Canyon sucks!">
```
Очень классное решение - графические символы передать в alt

efiand commented on index.html:
```diff
@@ -42,16 +42,16 @@
+    <main class="main-index">
+      <section class="hero">
+        <h2 class="visually-hidden">О Седона</h2>
+        <img src="#" alt="Пейзаж-гор">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#IMG-01

Это фон, который будет добавляться через CSS, потому что:
1. На нем нет какой-то предметной инфы, весьма безликие недетализированные виды.
2. На нем находится контентная текстовая эмблема, а перекрывают обычно фоны, но никак не смысловые изображения.
3. Если его изъять, смысл содержимого не пострадает.

efiand commented on index.html:
```diff
@@ -42,16 +42,16 @@
 
-    <h1 class="main-hero city-text">Седона — небольшой городок в Аризоне,
-      заслуживающий большего!</h1>
+       <h2 class="main-hero city-text">Седона — небольшой городок в Аризоне,
```
Этот текст по смыслу не похож на заголовок для какого-либо нижеследующего контента. Это скорее lead-текст. И он очень связан с причинами, поэтому ожидается в одной секции с ними.

Вот так было бы хорошо:
![изображение](https://user-images.githubusercontent.com/26799424/215150864-c3146dca-9ddc-4229-a367-b050a3125009.png)

efiand commented on index.html:
```diff
@@ -50,10 +50,40 @@ <h2 class="visually-hidden">О Седона</h2>
-
+      <p>Рассмотрим причины, по которым Седона круче, чем Гранд-Каньон!</p>
+      <section class="advantages">
+        <h2 class="visually-hidden">Преимущества</h2>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#ALLY-01
https://up.htmlacademy.ru/htmlcss/36/criteries#ALLY-02

Почему-то нет критерия про неинтерактивные доступные тексты, но принципы те же:

> Подпись в конце описания **должна** иметь точку. Точка нужна для интонационной паузы при чтении скринридером.

efiand commented on index.html:
```diff
@@ -50,10 +50,40 @@ <h2 class="visually-hidden">О Седона</h2>
+      <p>Рассмотрим причины, по которым Седона круче, чем Гранд-Каньон!</p>
+      <section class="advantages">
+        <h2 class="visually-hidden">Преимущества</h2>
+        <ul class="advantages-list">
```
Вот тут я вижу, что вы понимаете насчет имен классов, тут ведь не `advantages list`. Вот по этому принципу и надо переименовать все классы в шапке и hero

efiand commented on index.html:
```diff
@@ -50,10 +50,40 @@ <h2 class="visually-hidden">О Седона</h2>
+          <li class="advantages-item">
+            <h3 class="advantages-title">Настоящий городок</h3>
+            <p class="advantages-description">Седона — не аттракцион для туристов, там течёт своя жизнь</p>
+            <img src="#" alt="Город Седона">
```
Да, всё правильно, тут картинка контентная и относится именно к этой причине.

efiand replied:
https://up.htmlacademy.ru/htmlcss/36/criteries#ALLY-01

Во всех `alt` нужно проставить точки, если на конце текста нет другой пунктуации.

efiand commented on index.html:
```diff
@@ -50,10 +50,40 @@ <h2 class="visually-hidden">О Седона</h2>
+            <img src="#" alt="Город Седона">
+          </li>
+          <li class="advantages-item">
+            <h3 class="advantages-title">Небольшая
```
Лучше единообразно располагать короткие строчки, в данном случае:

```html
            <h3 class="advantages-title">Небольшая площадь</h3>
```

efiand commented on index.html:
```diff
@@ -84,6 +84,8 @@ <h3 class="advantages-title">Там есть
           </li>
         </ul>
       </section>
+      <h2 class="city-advantages">Приезжайте в Седону отдохнуть в комфорте и уюте!</h2>
```
Это, как и в примере с Преимуществами, рекомендуется расположить внутрь секции Рекомендаций в качестве лид- и обычного текста. Что касается имен классов - начинайте думать в том ключе, что `одинаковые вещи делаются одинаково`:

![изображение](https://user-images.githubusercontent.com/26799424/215153955-9e0eb874-ea19-4d26-8cb7-3b2032342bd7.png)

Классы не несут семантической роли, они нужны только для оформления, поэтому выгодно не создавать клонов одной и той же сущности, которая отличается только контентом.

TromiUss replied:
Вы хотите сказать, что текст с приглашением  в город лучше заключить в секцию с преимуществами?
Я просто посчитал, что это не относится к этой секции, да и в фигме блок со вторым текстом отделен от преимуществ.

Правильно я понимаю, что страничные блоки в фигме должны так же соблюдаться и в коде?

efiand replied:
> да и в фигме блок со вторым текстом отделен от преимуществ.

Он даже слегка (похоже на баг макета, реализуемый доп. классом)  ближе к списку карточек, чем в первом макете.
![изображение](https://user-images.githubusercontent.com/26799424/215313945-40f1ef6c-3a51-4a79-bc28-870cbe35bdaf.png)


В фигме нас интересуют только визуальные моменты. Видение дизайнера о семантике может отличаться от того, что нужно в вебе.

> страничные блоки в фигме должны так же соблюдаться и в коде?

Поэтому - смотря под каким углом взглянуть на ситуацию, дизайн мы соблюдаем, на техническую реализацию в фигме особо внимания не обращаем.

Мы видим с точки зрения дизайна три блока, у которых:
* Заголовок скрытый
* Слоган (по задаче, данной дизайнеру, это заголовок, наполненный незаголовочным контентом, и в реальном мире у нас всегда в ТЗ есть точное упоминание, заголовок это или слоган, но так как у нас только макет, мы допустим, что это окончательные тексты, хотя в реальности на момент создания макета тексты могут быть предварительными, теряющими актуальность к сдаче проекта, и мы на них не ориентируемся, там может быть хоть lorem ipsum)
* Описание
* Прочий контент, отличающийся в трех случаях.

Если посмотреть еще раз на контент второй секции, нет ощущения, что он уж совсем не подходит к нижерасположенному списку, не так ли? Да, возможно, связь не так очевидна, как в первой секции, но тем не менее - если она есть в первой, то пользователю единообразный вид поможет понять, что и здесь есть взаимосвязь. Это такой прием дизайна. Если бы ее не было, скорее всего, придумали бы что-то совершенно иное по оформлению.

И нет никаких причин делать одинаковые вещи разным образом. Хороший дизайн - (а в академии он неплох, скажу я вам как бакалавр дизайна) - всегда про то, что одинаковые вещи сделаны одинаково, чтобы пользователя не путать, а наоборот, помочь ему ориентироваться.
И при разработке мы тоже стараемся одинаково делать, то есть тут одинаковые блоки одинаково сверстать имеет смысл, с одинаковыми тегамии и классами.

efiand commented on index.html:
```diff
@@ -86,6 +86,28 @@ <h3 class="advantages-title">Там есть
+      <section class="recommendations">
+        <ul class="recommendations-list">
+          <li class="recommendations-item">
+            <img src="#" alt="Жильё">
```
Это декоративное изображение, так как на нем нет ничего конкретного. И в остальных пунктах так же.

efiand commented on index.html:
```diff
@@ -108,6 +108,13 @@ <h3 class="recommendations-title">Сувениры</h3>
         </ul>
       </section>
+      <section class="hotels-search">
+        <h2 class="hotels-search question">Заинтересовались?</h2>
```
Здесь опять дефисы у классов потеряны, см. advantages и recommendations как пример хорошего наименования.

efiand commented on index.html:
```diff
@@ -108,6 +108,13 @@ <h3 class="recommendations-title">Сувениры</h3>
+      <section class="hotels-search">
+        <h2 class="hotels-search question">Заинтересовались?</h2>
+        <p class="hotel-search proposal">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в Седоне</p>
+        <button type="button" > ПОИСК ГОСТИНИЦЫ В седоне
```
Большие буквы - это выражение экспрессии в тексте: отчаяние и т. п. Желательно с целью правильного произнесения читалками использовать правильный регистр. Дизайнер не указ того, как делать живой веб: если он на макете так сделал, это не повод воспроизводить. Обычно первая буква фразы большая, остальные - маленькие, за исключением особых случаев, например, название города Седона.

efiand commented on index.html:
```diff
@@ -108,6 +108,13 @@ <h3 class="recommendations-title">Сувениры</h3>
+        <h2 class="hotels-search question">Заинтересовались?</h2>
+        <p class="hotel-search proposal">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в Седоне</p>
+        <button type="button" > ПОИСК ГОСТИНИЦЫ В седоне
+          <!-- тут Popup форма -->
```
Явно не тут, не в кнопке.
Может быть после кнопки - но рекомендуется в целях меньшего количества перерисовок страницы браузером попапы размещать после футера страницы.

efiand commented on index.html:
```diff
@@ -116,6 +116,16 @@ <h2 class="hotels-search question">Заинтересовались?</h2>
-    <footer></footer>
+    <footer>
+      <section class="mailing">
+        <h2 class="mailing mailing-text">Подпишитесь на рассылку</h2>
```
Не нужно дублировать класс, содержащий стили для оформления секции, в дочерних элементах. `mailing-text` - уже достаточно

efiand commented on index.html:
```diff
@@ -116,6 +116,16 @@ <h2 class="hotels-search question">Заинтересовались?</h2>
+    <footer>
+      <section class="mailing">
+        <h2 class="mailing mailing-text">Подпишитесь на рассылку</h2>
+        <p class="mailing mailing-description">Только полезная информация и никакого спама, честное бойскаутское!</p>
```
(однако я бы при задании имен классов подумал о том, что это очень похоже на крупный и мелкий текст трех секций из main)

efiand commented on index.html:
```diff
@@ -116,6 +116,16 @@ <h2 class="hotels-search question">Заинтересовались?</h2>
+      <section class="mailing">
+        <h2 class="mailing mailing-text">Подпишитесь на рассылку</h2>
+        <p class="mailing mailing-description">Только полезная информация и никакого спама, честное бойскаутское!</p>
+        <form method="post" action="#">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-04

Для работоспособности форм давайте им всем зададим action `https://echo.htmlacademy.ru`

efiand commented on index.html:
```diff
@@ -116,6 +116,16 @@ <h2 class="hotels-search question">Заинтересовались?</h2>
+        <h2 class="mailing mailing-text">Подпишитесь на рассылку</h2>
+        <p class="mailing mailing-description">Только полезная информация и никакого спама, честное бойскаутское!</p>
+        <form method="post" action="#">
+          <input type="text" name="e-mail" value="Ваш e-mail">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-04

Единственное поле в форме всегда обязательно к заполнению.

efiand replied:
Так как надпись по дизайну прозрачна, а смысл ее напоминает не адрес, а подсказку, это не `value`, а `placeholder`

efiand commented on index.html:
```diff
@@ -116,6 +116,16 @@ <h2 class="hotels-search question">Заинтересовались?</h2>
+        <p class="mailing mailing-description">Только полезная информация и никакого спама, честное бойскаутское!</p>
+        <form method="post" action="#">
+          <input type="text" name="e-mail" value="Ваш e-mail">
+          <input type="submit" value="ПОДПИСАТЬСЯ">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-03

Используйте современный тег `<button>`, как в примере формы.

efiand commented on index.html:
```diff
@@ -116,6 +116,16 @@ <h2 class="hotels-search question">Заинтересовались?</h2>
+        <form method="post" action="#">
+          <input type="text" name="e-mail" value="Ваш e-mail">
+          <input type="submit" value="ПОДПИСАТЬСЯ">
+        </form>
```
И регистр текста, как везде, желательно писать нормативно (`Подписаться`).
Визуальную имитацию uppercase будем делать стилями.

efiand commented on index.html:
```diff
@@ -116,6 +116,16 @@ <h2 class="hotels-search question">Заинтересовались?</h2>
+        <h2 class="mailing mailing-text">Подпишитесь на рассылку</h2>
+        <p class="mailing mailing-description">Только полезная информация и никакого спама, честное бойскаутское!</p>
+        <form method="post" action="#">
+          <input type="text" name="e-mail" value="Ваш e-mail">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-05

Не сделано.

efiand commented on index.html:
```diff
@@ -121,6 +121,7 @@ <h2 class="hotels-search question">Заинтересовались?</h2>
         <h2 class="mailing mailing-text">Подпишитесь на рассылку</h2>
         <p class="mailing mailing-description">Только полезная информация и никакого спама, честное бойскаутское!</p>
         <form method="post" action="#">
+          <img src="#" alt="красивый вид на каньон">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#IMG-01

Напрасно. Этот красивый вид некрасивый: он сильно размыт и частично скрыт за текстом. Это классический декоративный фон, вставляется через CSS.

efiand commented on index.html:
```diff
@@ -126,7 +126,24 @@ <h2 class="mailing mailing-text">Подпишитесь на рассылку</h
         </form>
       </section>
-
+      <nav class="true-footer">
```
Навигация обычно содержит ключевые разделы основного сайта, а это - ссылки даже не на этот сайт. Это не `<nav>`. С целью построения сетки при стилизации в дальнейшем можно обойтись без этой обертки, либо добавить один `<div>`.

efiand replied:
`true-` не выполняет никаких оформительских задач. Можно не усложнять нейминг и оставить здесь класс `footer`, а тег `<footer>` может вообще жить без классов, ведь семантика и оформление - это разные задачи.

efiand commented on index.html:
```diff
@@ -126,7 +126,24 @@ <h2 class="mailing mailing-text">Подпишитесь на рассылку</h
+      <nav class="true-footer">
+        <ul class="true-footer social-network">
+          <li class="social-network social-network-item">
+            <a href="https://vk.com/htmlacademy" class="social-network social-network-link-wk"></a>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#PROJ-02

`-vk`

Называйте вещи своими именами, а то доказывайте потом, что это не транслит.

efiand replied:
https://up.htmlacademy.ru/htmlcss/36/criteries#ALLY-02

Внутрь каждой ссылки можно положить `<span class="visually-hidden">`.

efiand commented on index.html:
```diff
@@ -126,7 +126,24 @@ <h2 class="mailing mailing-text">Подпишитесь на рассылку</h
+            <a href="https://www.youtube.com/user/htmlacademyru" class="social-network social-network-link-YouTube"></a>
+          </li>
+        </ul>
+        <h2 class="phone-number">+7 (812) 812-12-12</h2>
```
Если мы делаем заголовок, то он озаглавливает текст, идущий после него, до следующего заголовка того же уровня. Телефон как заголовок ссылки на академию - это странно.

https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-03

> Для почты и телефона использованы схемы `mailto:` и `tel:`.

Это ссылка на телефон, а не заголовок.

efiand commented on index.html:
```diff
@@ -126,7 +126,24 @@ <h2 class="mailing mailing-text">Подпишитесь на рассылку</h
+        </ul>
+        <h2 class="phone-number">+7 (812) 812-12-12</h2>
+          <a href="#" class="html-academy-link">
+            <img src="#" alt="векторное изображение .html academy">
```
Название компании лучше писать не как в логотипе, а как на самом деле - `HTML Academy`. И более точное описание - что это не просто векторное изображение, а логотип.

efiand commented on index.html:
```diff
@@ -126,7 +126,24 @@ <h2 class="mailing mailing-text">Подпишитесь на рассылку</h
+          </li>
+        </ul>
+        <h2 class="phone-number">+7 (812) 812-12-12</h2>
+          <a href="#" class="html-academy-link">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML

> Если в макете есть требования, касающиеся адресов ссылок, они должны быть выполнены.

![изображение](https://user-images.githubusercontent.com/26799424/215168734-ad9a99f2-dfd8-4c4b-832e-9db367e8cf20.png)

efiand commented on index.html:
```diff
@@ -129,17 +129,17 @@ <h2 class="mailing mailing-text">Подпишитесь на рассылку</h
         <ul class="true-footer social-network">
           <li class="social-network social-network-item">
-            <a href="https://vk.com/htmlacademy" class="social-network social-network-link-wk"></a>
+            <a href="https://vk.com/htmlacademy" class="social-network social-network-link-wk">вк</a>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#ALLY-02

Эти подписи скрыты. Можно использовать для всех этих ссылок `<span class="visually-hidden">`

efiand reviewed (commented):
Критические проблемы, которые обязательно исправить для принятия задания, отмечаю реакцией 👎🏻 
Остальные комментарии - советы, как сделать лучше.

TromiUss committed:
Исправления

Классы, ссылки и заглушки, порядок атрибутов, точки в подписях, форматирование, блок хиро, заголовок, фоны,

TromiUss committed:
Исправления 2

Регистр, кнопка, <label>, фон, формы, Попап

TromiUss committed:
исправления 3

Телефон, Логотип

TromiUss committed:
экспорт

Экспорт фото

efiand commented on index.html:
```diff
@@ -1,74 +1,73 @@
+      <nav class="header-navigation">
+        <ul class="header-list">
+          <li class="header-item">
+            <a href="#!">Главная</a>
```
Это ссылка на `index.html`, допустимо без `href` на данной странице

efiand commented on index.html:
```diff
@@ -1,74 +1,73 @@
           </li>
-          <li class="header-page item">
-            <a href="#" class="navigation-link">
+          <li class="user-pitem">
```
Опечатка в имени класса

efiand commented on index.html:
```diff
@@ -1,74 +1,73 @@
               <span class="visually-hidden">
-                Избранное
-            </span>
+                Избранное.
```
По-прежнему не хватает контентной информации о количестве закладок, например:

```html
<a class="user-link" href="#!">
  <span class="visually-hidden">Избранное:</span>
  <span class="user-count">12</span>
</a>
```

efiand commented on index.html:
```diff
@@ -1,74 +1,73 @@
       </nav>
-    </header>
-    <main class="main-index">
       <section class="hero">
```
Эта информация встречается только на главной странице, и ее стоит положить в `<main>`

efiand commented on index.html:
```diff
@@ -1,74 +1,73 @@
       <section class="hero">
-        <h2 class="visually-hidden">О Седона</h2>
-        <img src="#" alt="Пейзаж-гор">
+        <h1 class="visually-hidden">О Седона.</h1>
```
Название склоняется

efiand commented on index.html:
```diff
@@ -1,74 +1,73 @@
+        заслуживающий большего!</h2>
       <p>Рассмотрим причины, по которым Седона круче, чем Гранд-Каньон!</p>
+    </header>
+    <main class="main-index"></main>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-07

Лишнее закрытие тега `<main>`.

Текст `Рассмотрим причины` и список этих самых причин - хочется видеть в одной секции.

efiand commented on index.html:
```diff
@@ -1,74 +1,73 @@
-
-      <h2 class="main-hero city-text">Седона — небольшой городок в Аризоне,
-      заслуживающий большего!</h2>
+      <h2 class="city-text">Седона — небольшой городок в Аризоне,
```
Этот текст не подходит на роль заголовка, это скорее параграф или `<b>`, а заголовок для этого контента уже есть, скрытый.

В общем, чтобы долго не писать, в виде кода покажу, как это могло бы выглядеть с учетом всех нюансов семантики и стилевого единообразия:

![изображение](https://user-images.githubusercontent.com/26799424/215559565-08210662-af84-4354-88af-a13d3c3597ea.png)

efiand commented on index.html:
```diff
@@ -89,60 +88,62 @@ <h2 class="city-advantages">Приезжайте в Седону отдохну
         <ul class="recommendations-list">
           <li class="recommendations-item">
-            <img src="#" alt="Жильё">
+            <img src="#" alt="Жильё.">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#IMG-01

Это декоративные изображения, их добавлять в разметку не нужно.

efiand commented on index.html:
```diff
@@ -89,60 +88,62 @@ <h2 class="city-advantages">Приезжайте в Седону отдохну
+        <h2 class="hotels-question">Заинтересовались?</h2>
+        <p class="hotel-proposal">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения
+          гостиниц в Седоне</p>
+        <button type="button"> Поиск гостиницы в Седоне
```
https://up.htmlacademy.ru/htmlcss/36/criteries#PROJ-01

И все-таки хочется единообразного форматирования.
Сравните две картинки:

1. Как сейчас
![изображение](https://user-images.githubusercontent.com/26799424/215560846-1cdccce4-eeb1-496a-8c8b-66a1dacd236c.png)

2. После единообразного форматирования
![изображение](https://user-images.githubusercontent.com/26799424/215561957-f7142ee6-99d3-4f3d-9ae3-8800691a5432.png)

efiand commented on index.html:
```diff
@@ -112,17 +109,18 @@ <h2 class="hotels-question">Заинтересовались?</h2>
-          <input type="text" name="e-mail" value="Ваш e-mail">
-          <input type="submit" value="Подписаться">
+        <form method="post" action="https://echo.htmlacademy.ru">
+          <label>
```
А где текст?

efiand commented on index.html:
```diff
@@ -112,17 +109,18 @@ <h2 class="hotels-question">Заинтересовались?</h2>
-          <input type="submit" value="Подписаться">
+        <form method="post" action="https://echo.htmlacademy.ru">
+          <label>
+            <input type="text" name="e-mail" placeholder="Ваш e-mail" required>
```
Лучше type="email"

efiand commented on index.html:
```diff
@@ -112,17 +109,18 @@ <h2 class="hotels-question">Заинтересовались?</h2>
+          <label>
+            <input type="text" name="e-mail" placeholder="Ваш e-mail" required>
+          </label>
+          <button type="button">Подписаться</button>
```
тип кнопки - submit

efiand commented on index.html:
```diff
@@ -88,16 +88,19 @@ <h2 class="city-advantages">Приезжайте в Седону отдохну
     <section class="recommendations">
       <ul class="recommendations-list">
         <li class="recommendations-item">
+          <img src="images/living.svg" width="75" height="72" alt="">
```
Решили, что это декорации, и даже удалили, и вот опять (

efiand commented on index.html:
```diff
@@ -139,7 +142,7 @@ <h2 class="mailing-text">Подпишитесь на рассылку</h2>
       <a href="tel:+78128121212">+7 (812) 812-12-12</a>
       <a href="https://htmlacademy.ru" class="html-academy-link">
-        <img src="#" alt="Логотип HTML Academy">
+        <img src="images/HTML-logo.svg" width="114.9" height="32.9" alt="Логотип HTML Academy">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#PROJ-03
Прописные буквы в названии

TromiUss committed:
Недочеты

Убрал недочеты, следуя Вашим советам. Стилевое единообразие

efiand commented on index.html:
```diff
@@ -31,113 +31,99 @@
-              <span class="visually-hidden">
-                Избранное.
-              </span>
+              <span class="visually-hidden">Избранное.</span>
```
Не принципиально, но интонационно лучше будет именно здесь использовать двоеточие, ведь это как бы одна фраза `Избранное: 12`, спаны читалкой игнорируются

efiand commented on index.html:
```diff
@@ -31,113 +31,99 @@
+        <h2 class="visually-hidden">Сервисы.</h2>
+        <b class="content-title">Приезжайте в Седону отдохнуть в комфорте и уюте!</b>
+        <p class="content-text">Опытный персонал и качественное обслуживание!</p>
+        <ul class="servisec-list">
```
Тут мы немного упустили, но логичнее видеть `services` вместо `servisec`. Проблема глубже чем кажется: другой разработчик, либо вы спустя время, где-нибудь в связанном коде, в тех же стилях, напишете правильное название, ведь это ожидаемо, и будет ошибка. Вы ее, конечно, в итоге найдете, но потерянное на ровном месте время.

efiand commented on index.html:
```diff
@@ -31,113 +31,99 @@
+        </ul>
+      </section>
+      <section class="hotels-search">
+        <h2 class="visually-hidden"></h2>
```
Нужно как-то назвать

efiand commented on index.html:
```diff
@@ -31,113 +31,99 @@
+      <section class="hotels-search">
+        <h2 class="visually-hidden"></h2>
+        <b class="hotels-question">Заинтересовались?</b>
+        <p class="hotel-proposal">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в Седоне</p>
```
Допустим, что тут по какой-то причине мы не можем использовать те же классы, что и в других похожих местах, но даже для уникальных блоков рекомендуется выдерживать наименование в общем стиле, так проще потом ориентироваться. Например, вся секция `hotels-search`, слоган `hotels-question`, и при работе с этим блоком мы ожидаем `hotels-proposal`, а не `hotel-proposal`. Даже если мы качественно тестируем свой результат, все равно на такой ерунде можно потерять время, а на весь проект таких потерь может набежать существенно. Поэтому стараемся писать более единообразные и упорядоченные решения.

efiand reviewed (commented):
Отличная работа, стало намного лучше, почти нет критических замечаний (одно, да вот вчера [это](https://github.com/htmlacademy-htmlcss/2300497-sedona-36/pull/2#discussion_r1090991227) забыл отметить. Надо переименовать картинку в `htmlacademy-logo` - лучше так, потому что:

- Имя, в которм изменен `только` регистр букв, не будет восприниматься гитом как изменение, и это не получется запушить.
- Называем вещи своими именами, это не логотип языка разметки

efiand commented:
Поскольку проверка повторная, а замечаний почти не осталось, можно их перенести на следующее задание.
